core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2021,20 +2057,6 @@ wasmtime_ssp_path_open(
         return error;
     }
 
-    {
```

Comment:
I'm not sure if this block is needed for anything else - it seems like it's giving unnecessary rights to the fd. It was causing this test [path_open_read_without_rights](https://github.com/WebAssembly/wasi-testsuite/blob/main/tests/rust/src/bin/path_open_read_without_rights.rs#L31-L34) to fail.

Reply:
Even though that perfectly matches the standard, I wonder if that won't introduce any backward compatibility within WAMR - @wenyongh / @xwang98 I wonder how should we handle this situations? Have we experienced such things in the past? Would it be enough to highlight this incompatibility in release notes?

Reply:
In my memory we didn't experience such things in libc-wasi, but agree to highlight this incompatibility in release notes. It may be a security issue to give unnecessary rights to the fd, we just fix it to match the standard.

Reply:
@wenyongh thanks for confirming, I've moved this change to a separate PR, https://github.com/bytecodealliance/wasm-micro-runtime/pull/2579